src/PluginsManager/Microsoft.Performance.Toolkit.Plugins.Runtime/Manager/IPluginsManager.cs

```diff
@@ -142,21 +165,29 @@ public interface IPluginsManager
         /// <param name="pluginPackagePath">
         ///     The path to the plugin package to be installed.
         /// </param>
-        /// <param name="overwriteInstalled">
-        ///     Whether to overwrite the existing plugin if there's one isntalled. 
-        /// </param>
         /// <param name="cancellationToken">
         ///     Signals that the caller wishes to cancel the operation.
         /// </param>
         /// <param name="progress">
         ///     Indicates the progress of plugin installation.
         /// </param>
         /// <returns>
-        ///     <c>true</c> if the plugin has been successfully installed. <c>false</c> otherwise.
+        ///     The <see cref="InstalledPluginInfo"/> if plugin is successfully installed.. <c>null</c> otherwise.
         /// </returns>
-        Task<bool> InstallLocalPluginAsync(
+        /// <exception cref="ArgumentNullException">
+        ///     Throws when <paramref name="pluginPackagePath"/> is <c>null</c>.
+        /// </exception>
+        /// <exception cref="PluginRegistryException">
+        ///     Throws when there is an error interacting with plugin registry.
```

Comment:
I would prefer if these errors were more fine-grained. Some exceptions can be recovered from, others cannot.

For example, if the issue is that the plugin registry is corrupt, we can recover by simply deleting the registry and starting over. But, if the issue is that the registry cannot be accessed, then we have a bigger problem since we need to change its location